homeassistant/auth/providers/command_line.py

```diff
@@ -68,6 +68,7 @@ async def async_validate_login(self, username: str, password: str) -> None:
                 *self.config[CONF_ARGS],
                 env=env,
                 stdout=asyncio.subprocess.PIPE if self.config[CONF_META] else None,
+                close_fds=False,  # required for posix_spawn
```

Comment:
The problem is that this will break any 3rd party lib using subprocess

Reply:
What am I missing?  Are there specific fds open that will be a problem for the subprocess in this instance?

> If close_fds is true, all file descriptors except 0, 1 and 2 will be closed before the child process is executed. Otherwise when close_fds is false, file descriptors obey their inheritable flag as described in [Inheritance of File Descriptors](https://docs.python.org/3/library/os.html#fd-inheritance).

> A file descriptor has an “inheritable” flag which indicates if the file descriptor can be inherited by child processes. Since Python 3.4, file descriptors created by Python are non-inheritable by default.

> On UNIX, non-inheritable file descriptors are closed in child processes at the execution of a new program, other file descriptors are inherited.

Reply:
Some more back story. https://github.com/python/cpython/issues/86904

I don’t think we are crossing a security boundary here (but maybe I’m not seeing it). Most of these invocations are even using a shell directly so I’d expect anything they are executing to run at the same privilege level as the python process.

Reply:
Sorry, I meant in general. If we have to update all our calls to pass `close_fds=False`, then 3rd party libs calling a sub process would have to do the same.